compiler/backend/src/org/jetbrains/kotlin/codegen/MemberCodegen.java

```diff
@@ -108,6 +109,8 @@ public MemberCodegen(
         this.propertyCodegen = new PropertyCodegen(context, v, functionCodegen, this);
         this.parentCodegen = parentCodegen;
         this.jvmAssertFieldGenerated = false;
+        // TODO fix the script runner (shouldn't require a SMAP) & don't write trivial SMAPs to script main classes.
+        this.alwaysWriteSourceMap = this instanceof ScriptCodegen;
```

Comment:
Could you clarify this a bit? What exception is failed?

Reply:
See the example in the PR/commit description - if the SMAP for the file class does not exist, the stack trace will contain "Unknown Source" instead of actual line numbers. There's actually a test that checks the stack trace https://github.com/JetBrains/kotlin/blob/master/compiler/testData/integration/smoke/scriptException/script.kts but it happens to pass because `error("my error")` is an inline call which forces generation of a SMAP; replacing it with `throw ...` breaks the test.

Reply:
`smaps` triggers `visitSource` call, and looks like it's not called in general case for scripts

Reply:
so debug information is missed

Reply:
Oh, so it's not that it always tries to use the SMAP, it's just that ScriptCodegen.generateDeclaration should've called visitSource...that makes more sense, thanks.